source/plugin/data/custom/datamanipulators.rst

```diff
@@ -91,15 +93,142 @@ the work for you. In addition to the above, you need to override the following m
+    import org.spongepowered.api.data.value.mutable.Value;
+
+    import java.util.Optional;
+
+    import com.example.homes.Home;
+    import com.example.homes.ImmutableHomeData;
+    import com.example.homes.MyKeys;
+
+    public class HomeDataImpl extends AbstractSingleData<Home, HomeData, ImmutableHomeData> implements HomeData {
+        public HomeDataImpl(Home value) {
+            super(value, MyKeys.DEFAULT_HOME);
+        }
+
+        // It's best to provide an empty constructor with "default" values
+        public HomeDataImpl() {
+            this(null);
+        }
+
+        // Override if you have a separate interface
+        @Override
+        public Value<Home> defaultHome() {
+            return getValueGetter();
+        }
+
+
+        @Override
+        public int getContentVersion() {
+            // Update whenever the serialization format changes
+            return 1;
+        }
+
+        @Override
+        protected Value<Home> getValueGetter() {
+            /* We recommend making this a field if implementing this immutable version,
+               as it will never change */
+            return Sponge.getRegistry().getValueFactory()
+                    .createValue(MyKeys.DEFAULT_HOME, getValue(), null);
+        }
+        
+        // asMutable() when implementing immutable data
+        @Override
+        public ImmutableHomeData asImmutable() {
+            return new ImmutableHomeDataImpl(getValue());
+        }
+
+        // Only required on mutable implementation
+        @Override
+        public Optional<HomeData> fill(DataHolder dataHolder, MergeFunction overlap) {
+            HomeData merged = overlap.merge(this, dataHolder.get(HomeData.class).orElse(null));
+            setValue(merged.getValue());
+
+            return Optional.of(this);
+        }
+
+        // Only required on mutable implementation
+        @Override
+        public Optional<HomeData> from(DataContainer container) {
+            if(container.contains(MyKeys.DEFAULT_HOME)) {
+                // Loads the structure defined in toContainer
+                setValue(container.getSerializable(MyKeys.DEFAULT_HOME.getQuery(), Home.class).get());
+                return Optional.of(this);
+            }
+            return Optional.empty();
+        }
+
+        // Only required on mutable implementation
+        @Override
+        public HomeData copy() {
+            return new HomeData(getValue());
+        }
+
+        @Override
+        public DataContainer toContainer() {
+            DataContainer container = super.toContainer();
+            // This is the simplest, but use whatever structure you want!
+            container.set(MyKeys.DEFAULT_HOME.getQuery(), getValue());
+            return container;
+        }
+    }
+
 
 Compound Types
 ==============
 
+Whereas Single types only support one value, compound types support however many values you want. The downside, however,
```

Comment:
Please be consistent with capitalisation. If Single type is capitalised then so should Compound type.